lib/winapi/sync.c

```diff
@@ -1,11 +1,145 @@
 #include <synchapi.h>
 #include <assert.h>
+#include <stdatomic.h>
 #include <stdbool.h>
 #include <processthreadsapi.h>
 #include <winbase.h>
 #include <winerror.h>
 #include <xboxkrnl/xboxkrnl.h>
 
+static BOOL WINAPI condvar_init (PINIT_ONCE InitOnce, PVOID Parameter, PVOID *Context)
+{
+    PCONDITION_VARIABLE ConditionVariable = Parameter;
+    NTSTATUS status;
+    status = NtCreateEvent(&ConditionVariable->eventHandles[0], NULL, SynchronizationEvent, FALSE);
+    assert(NT_SUCCESS(status));
+    status = NtCreateEvent(&ConditionVariable->eventHandles[1], NULL, NotificationEvent, FALSE);
+    assert(NT_SUCCESS(status));
+    return true;
+}
+
+VOID InitializeConditionVariable (PCONDITION_VARIABLE ConditionVariable)
+{
+    ConditionVariable->initOnce.Ptr = 0;
+    ConditionVariable->eventHandles[0] = INVALID_HANDLE_VALUE;
+    ConditionVariable->eventHandles[1] = INVALID_HANDLE_VALUE;
+    atomic_store(&ConditionVariable->waitCount, 0);
```

Comment:
Why isn't this being done as part of `condvar_init`?

I assume the intention of the InitOnce was so that people could use ConditionVariables even when using `CONDITION_VARIABLE_INIT`? But then you currently don't initialize `ConditionVariable->waitCount`?

Reply:
> I can't follow here

I have no idea what I meant here either. I'll just mark this conversation as resolved.

I agree with your response. `waitCount` will be initialized by `CONDITION_VARIABLE_INIT` (or by `InitializeConditionVariable` on the linked line).